src/popup.ts

```diff
@@ -0,0 +1,414 @@
+type SavedPrompt = {
+    text: string;
+    tags: string[];
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+    const addPromptButton = document.getElementById('add-prompt') as HTMLButtonElement;
+    const copyPromptsButton = document.getElementById('copy-prompts') as HTMLButtonElement;
+    const toggleTagFilterButton = document.getElementById('toggle-tag-filter') as HTMLButtonElement;
+    const promptInput = document.getElementById('prompt-input') as HTMLInputElement;
+    const tagFilterSelect = document.getElementById('tag-filter-select') as HTMLSelectElement;
+    const promptList = document.getElementById('prompt-list') as HTMLDivElement;
+    const tagFilterSection = document.getElementById('tag-filter-section') as HTMLDivElement;
+    const kebabMenuButton = document.getElementById('kebab-menu-button') as HTMLButtonElement;
```

Comment:
Most of the changes are in this file, converting parts of it piecemeal to typescript